### PR TITLE
snyk ms:20170412 performance optimization

### PR DESCRIPTION
**Pull request by Delagen**
- dramatically performance optimization
- added test for Infinity and long strings performance
- standard style fix, fix missing devDependencies

### CODE REVIEW

**Comment by leo** [comment]
Looks good to me! But can you please remove `standard` and replace it with `xo`? Or simply leave it out and we'll add it later.

**Comment by Delagen** [comment]
What you don't like in standard? May be left only eslint with fixers!?

**Comment by leo** [comment]
Because we have XO on almost all of our repositories... 😊

**Comment by Delagen** [comment]
replaced by eslint-config-xo which IMHO better than second linter

**Comment by Delagen** [comment]
Any review?

**Comment by leo** [comment]
Will have a look soon! Currently pretty busy.

**Comment by thevtm** [comment]
Hi!
I did a little benchmarking and found out that the original implementation seems to be around 2x faster than the PR implementation.

[Benchmark source](https://gist.github.com/thevtm/063a834083a42af535e26196a5d3c8d5)

```
original x 350,875 ops/sec ±1.68% (84 runs sampled)
new x 130,267 ops/sec ±1.02% (87 runs sampled)
Fastest is original
Done in 12.50s.
```

**Comment by Delagen** [comment]
@thevtm Just insert one test value `'1'.repeat(29) + 'Z'` to invert results

**Comment by thevtm** [comment]
@Delagen thanks! I haven't considered that case.
I ended up finding a better solution.
Changing the Regex from:
 `/^((?:\d+)?\.?\d+) *(milliseconds?|msecs?|ms|seconds?|secs?|s|minutes?|mins?|m|hours?|hrs?|h|days?|d|years?|yrs?|y)?$/i`
to:
`/^((?:\d+)?\.?\d+) *(\w{0,12})?$/i`
seems to produce better results in all cases.

[Benchmark source](https://gist.github.com/thevtm/063a834083a42af535e26196a5d3c8d5)

```
original x 38,472 ops/sec ±2.05% (88 runs sampled)
new x 39,713 ops/sec ±1.89% (82 runs sampled)
mine x 74,820 ops/sec ±1.84% (84 runs sampled)
Fastest is mine
Done in 17.97s.
```

**Comment by Delagen** [comment]
@thevtm thanks for benchmarking, updated code with your regexp

**Comment by knoxcard** [comment]
You guys are amazing thanks, look forward to pulling the latest!

File: index.js [hunk 1]
```diff
@@ -8,33 +10,40 @@ var h = m * 60;
...
- *
- *  - `long` verbose formatting [false]
- *
- * @param {String|Number} val
- * @param {Object} [options]
- * @throws {Error} throw an error if val is not a non-empty string or a number
- * @return {String|Number}
- * @api public
- */
+var scales = new Map(
```
Comment (TooTallNate) [inline index.js]: This module is meant to be run in a variety of JavaScript environments, including (possibly older) web browsers. I don't think we'll be able to use `Map` here.

**Comment by Delagen** [comment]
@TooTallNate if you about IE8, It's dead, even webpack removed support of it

**Comment by ForbesLindesay** [comment]
@Delagen Even IE10 doesn't support Map. IE11's support is pretty hit and miss (although it probably supports everything you need here): https://kangax.github.io/compat-table/es6/

**Comment by Delagen** [comment]
@ForbesLindesay Modern development use a couple bunch of polyfills or transpilers when target IE, so it's not a problem at all. This PR is about 1 year old, so it can never merge.

**Comment by rauchg** [comment]
Interested in this, but PR has to be re-factored

**Pull request closed without merging**